flocker/control/test/test_protocol.py

```diff
@@ -586,25 +611,30 @@ def test_nodestate_notifies_all_connected(self):
         desired configuration.
         """
         self.control_amp_service.configuration_service.save(TEST_DEPLOYMENT)
-        self.protocol.makeConnection(StringTransport())
-        another_protocol = ControlAMP(self.reactor, self.control_amp_service)
-        another_protocol.makeConnection(StringTransport())
-        sent1 = []
-        sent2 = []
 
-        self.patch_call_remote(sent1, self.protocol)
-        self.patch_call_remote(sent2, protocol=another_protocol)
+        agents = [FakeAgent(), FakeAgent()]
+        clients = list(AgentAMP(Clock(), agent) for agent in agents)
```

Comment:
It doesn't really matter either way, but I think since each of these represents a notionally different computer it makes sense for them to have different clocks.